modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/address/storage/AddressStorage.scala

```diff
@@ -1,67 +1,32 @@
 package org.tessellation.dag.l1.infrastructure.address.storage
 
-import cats.effect.MonadCancelThrow
+import cats.effect.Async
+import cats.effect.kernel.Ref
 import cats.syntax.functor._
-import cats.syntax.traverse._
 
 import org.tessellation.dag.l1.domain.address.storage.AddressStorage
-import org.tessellation.dag.l1.infrastructure.db.Database
 import org.tessellation.schema.address.Address
 import org.tessellation.schema.balance.Balance
 
-import doobie.implicits._
-import doobie.quill.DoobieContext
 import eu.timepit.refined.auto._
-import io.getquill.context.Context
-import io.getquill.{Literal, SqliteDialect}
 
 object AddressStorage {
-  case class StoredAddress(address: Address, balance: Balance)
+  def make[F[_]: Async]: F[AddressStorage[F]] =
+    Ref.of[F, Map[Address, Balance]](Map.empty).map(make(_))
 
-  type Ctx = DoobieContext.SQLite[Literal] with Context[SqliteDialect, Literal]
-
-  def make[F[_]: MonadCancelThrow: Database]: AddressStorage[F] =
-    make[F](new DoobieContext.SQLite[Literal](Literal))
-
-  def make[F[_]: MonadCancelThrow](ctx: Ctx)(implicit db: Database[F]): AddressStorage[F] =
+  def make[F[_]: Async](balances: Ref[F, Map[Address, Balance]]): AddressStorage[F] =
```

Comment:
Maybe we can clean some dependencies from build.sbt at this point as well?

Reply:
Also there were decoder/encoder pairs created for `Balance` for sure, probably also for `Address` and other data. We can clean it as well I think.

Reply:
I think we cannot because Database currently is used i.e. by trust, but I'll double check that. Anyway, I'd keep that in a separate pull request